org.adempiere.base/src/org/compiere/model/MRule.java

```diff
@@ -331,4 +347,11 @@ public MRule markImmutable() {
 		return this;
 	}
 
+	@Override
+	protected boolean afterSave(boolean newRecord, boolean success)
+	{
+		Core.invalidateCompiledScript(getAD_Rule_ID());

```

Comment:
this shouldn't be needed, reset of cache for updated record should happens without any additional code.